### PR TITLE
throw error in mergeWith method if no merger is passed (#1503)

### PR DESCRIPTION
**Pull request by Brantron**
This ensures users get an informative error when calling `mergeWith` without a merge function

Details can be found at https://github.com/facebook/immutable-js/issues/1503.

### CODE REVIEW

**Comment by Brantron** [comment]
@facebook-github-bot Just signed CLA

File: src/functional/merge.js [hunk 1]
```diff
@@ -38,7 +38,8 @@ export function mergeWithSources(collection, sources, merger) {
     );
   }
   if (isImmutable(collection)) {
-    return collection.mergeWith
+    const shouldMergeWith = collection.mergeWith && merger instanceof Function;
```
Comment (asazernik) [inline src/functional/merge.js]: This logic translates all calls to `Immutable.merge(collection, sources)` into calls to `collection.concat(...sources)`. Should instead check for `collection.merge` if `merger` isn't a function. Something like:

```javascript
if (collection.mergeWith && merger instanceOf Function) {
  return collection.mergeWith(merger, ...sources);
} else if (collection.merge && !(merger instanceOf Function) { // REALLY not sure about this condition
  return collection.merge(...sources);
} else {
  return collection.concat(...sources);
}

**Comment by Brantron** [comment]
@asazernik thanks for taking a peak, pushed a commit.

**Comment by leebyron** [comment]
Squash merged in 2ea44e0ed31206de90d9ae6fc8082d63a052e8a3, thanks!

**Pull request closed without merging**